### PR TITLE
(fleet) support installing compliant datadog packages in the updater

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

This PR adds support for installing Datadog Packages in the updater.

### Motivation

We want to open the door to storing datadog packages in OCI compliant registries (gcr.io & co). The previous package format and code handling that format was not fully compliant with the OCI spec and would not have allowed us to easily migrate to OCI registires.

### Additional Notes

This PR also adds unit tests for all the updater code (except `local_api`).

### CODE REVIEW

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -153,6 +156,11 @@ func (u *updaterImpl) Stop(_ context.Context) error {
 func (u *updaterImpl) bootstrapStable(ctx context.Context) error {
 	u.m.Lock()
 	defer u.m.Unlock()
+	// both tmp and repository paths are checked for available disk space in case they are on different partitions
```
Comment (raphaelgavache) [inline pkg/updater/updater.go:159]: nice

File: pkg/updater/download_test.go [hunk 2]
```diff
@@ -6,322 +6,132 @@
...
+	assertEqualFS(t, s.PackageFS(fixtureSimpleV1), os.DirFS(tmpDir))
 }
 
-func (suite *DownloadTestSuite) TestDownloadCheckSize() {
-	t := suite.T()
-	downloader := newDownloader(suite.server.Client())
-	downloadPath := path.Join(suite.dir, testDownloadDir)
-	err := os.MkdirAll(downloadPath, 0755)
-	assert.NoError(t, err)
+func TestDownloadInvalidHash(t *testing.T) {
```
Comment (coignetp) [inline pkg/updater/download_test.go:128]: **🥜 nitpick:** ‏Can you add an invalid platform test?

**Review by coignetp — APPROVED** [review]

File: pkg/updater/download_test.go (hunk 2 above)
Comment (arbll) [reply pkg/updater/download_test.go:134]: done

File: pkg/updater/download.go [hunk 3]
```diff
@@ -42,149 +45,111 @@ func newDownloader(client *http.Client) *downloader {
...
 	reader := io.TeeReader(
-		io.LimitReader(resp.Body, maxArchiveSize),
+		io.LimitReader(resp.Body, ociLayoutArchiveMaxSize),
 		hashWriter,
 	)
-	archivePath := filepath.Join(tmpDir, archiveName)
-
-	// Check there is enough space to write the compressed archive
-	enoughSpace, err := tools.CheckAvailableDiskSpace(filepath.Dir(archivePath), uint64(pkg.Size))
+	ociLayoutArchivePath := filepath.Join(tmpDir, ociLayoutArchiveName)
```
Comment (BaptisteFoy) [inline pkg/updater/download.go:83]: **🥜 Nitpick:** ‏Only used once, we could directly set it in the create

**Review by BaptisteFoy — APPROVED** [review]

**Pull request merged**